atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -1214,6 +1210,12 @@ public PersistedSnapshot getCurrentSnapshot() {
     return currentSnapshot;
   }
 
+  public void setCurrentSnapshot(final PersistedSnapshot snapshot) {
```

Comment:
❓ Is this now called twice?

Reply:
In the case of snapshot replication, yes - we can't really avoid it. And we need the listener for the case where the local node takes a snapshot (i.e. transient snapshots) =/

I think that's acceptable considering it's not the hot path, and it doesn't do much.

Reply:
Maybe @deepthidevaki has an opinion on this since Nicolas is on FTO. 

We now call `setCurrentSnapshot` twice, one mostly directly and one later when the snapshot store persists the snapshots. Is this an issue during compacting?

Reply:
I don't think there is any correctness issue in calling it twice. It is just unnecessary work.